docs/source/api/apollo-server.mdx

```diff
@@ -599,6 +599,69 @@ The `start` method triggers the following actions:
 1. If your server is a [federated gateway](https://www.apollographql.com/docs/federation/managed-federation/overview/), it attempts to fetch its schema. If the fetch fails, `start` throws an error.
 2. Your server calls all of the [`serverWillStart` handlers](../integrations/plugins-event-reference/#serverwillstart) of your installed plugins. If any of these handlers throw an error, `start` throws an error.
 
+
+## `executeOperation`
+
+The async `executeOperation` method is used primarily for [testing GraphQL operations](../testing/testing/#testing-using-executeoperation) through the server's request pipeline without going through a full HTTP operation.
+
+The `executeOperation` method takes two arguments:
+* The first argument is a configuration options object
+* The second argument will be passed in to the current `ApolloServer` instance's `context` function
+
+Below are the available configuration options for the first argument of `executeOperation`:
+
+<table class="field-table">
+  <thead>
+    <tr>
+      <th>Name</th>
+      <th>Description</th>
+    </tr>
+  </thead>
+
+<tbody>
+<tr class="required">
+<td>
+
+##### `query`
+
+`String` or `<DocumentNode>`
+</td>
+
+<td>
+
+The GraphQL operation to be run. Note you must use the `query` key even if the operation is a mutation.
+</td>
+</tr>
+
+<tr>
+<td>
+
+##### `variables`
+
+`Object`
+</td>
+<td>
+
+You can optionally provide variables to be passed in as arguments to the `query`.
+</td>
+</tr>
+
+<tr>
+<td>
+
+##### `operationName`
+
+`String`
+</td>
+<td>
+
+Optional, but if included the `operationName` must be present in the provided `query`.
```

Comment:
Ah yeah so this one we should probably explain a little more, here's some extra context:

The document you provide for `query` might include the definitions of _multiple_ operations, like:

```graphql
query DoThing {
  thing
}

query DoOtherThing {
  otherThing
}
```

If you provide this document as `query`, Apollo Server _needs_ to know which of those queries you actually want to run, and `operationName` indicates this (so it'd be `DoThing` or `DoOtherThing`). If you _don't_ indicate this, Apollo Server throws an error during the step in the request pipeline where it attempts to "resolve" the operation (this is a confusingly named step that is not related to _resolvers_. In this case, "resolve" just means "figure out which operation to run").

We don't need to say all that, just need to say something like: "If `query` contains more than one operation definition, you must provide this option to indicate which operation to execute."

Reply:
cc @glasser to make sure I got this right

Reply:
Ah right as you document below it probably doesn't _throw_ an error in the case above but rather returns an `error` in the response